components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -2968,11 +2968,22 @@ def __init__(self, descriptor, toJSONMethod):
     def definition_body(self):
         ret = ''
         interface = self.descriptor.interface
+
         for m in interface.members:
             if m.isAttr() and not m.isStatic() and m.type.isJSONType():
                 name = m.identifier.name
+                conditions = MemberCondition(None, None, m.exposureSet)
+                ret_conditions = 'vec![' + ",".join(conditions) + "]"
                 ret += fill(
                     """
+                    let conditions = ${conditions};
+                    if !conditions.iter().any(|c|
+                         c.is_satisfied(
+                           SafeJSContext::from_ptr(cx),
+                           HandleObject::from_raw(obj),
+                           HandleObject::from_raw(obj))) {
+                      return false;
```

Comment:
I believe the timeouts in the tests are caused by returning here if there condition is not satisfied, rather than only executing the following code if the condition is satisfied.

Reply:
Thanks for review and checking. Do you mean that we should skip like below instead of returning false?

```rust
if flag = conditions.iter().any(|c|
         c.is_satisfied(
           SafeJSContext::from_ptr(cx),
           HandleObject::from_raw(obj),
           HandleObject::from_raw(obj)));
if flag {
    rooted!(in(cx) let mut temp = UndefinedValue());
    if !get_timing(cx, obj, this, JSJitGetterCallArgs { _base: temp.handle_mut().into() }) {
    return false;
    }
    if !JS_DefineProperty(cx, result.handle().into(),
                          b"timing\0" as *const u8 as *const libc::c_char,
                        temp.handle(), JSPROP_ENUMERATE as u32) {
    return false;
    }
}
```

I've tried it. But I still got timeout.
I'll continue to look into it.

Reply:
Ah, another problem is that we should be passing a global object as the third argument of is_satisfied based on https://doc.servo.org/script/dom/bindings/guard/enum.Condition.html#method.is_satisfied. We should be able to use something like:
```rust
let incumbent_global = GlobalScope::incumbent().expect("no incumbent global");
let global = incumbent_global.reflector().get_jsobject();
```
And then you could pass `global` as the third argument.

Reply:
Oh, sorry. I got it.
I fixed it and confirmed if the timeout isn't occurred.